src/modules/sitemap/services/sitemap.service.ts

```diff
@@ -0,0 +1,207 @@
+
+import { ContentPagesService, DataService } from '@meemoo/admin-core-api';
+import { Injectable, InternalServerErrorException } from '@nestjs/common';
+import moment from 'moment';
+
+import { SitemapConfig, SitemapItemInfo } from '../sitemap.types';
+
+import { VisitorSpaceStatus } from '~generated/database-aliases';
+import {
+	GetSitemapConfigDocument,
+	GetSitemapConfigQuery,
+} from '~generated/graphql-db-types-hetarchief';
+import { AssetsService } from '~modules/assets/services/assets.service';
+import { AssetFileType } from '~modules/assets/types';
+import { IeObjectLicense } from '~modules/ie-objects/ie-objects.types';
+import { IeObjectsService } from '~modules/ie-objects/services/ie-objects.service';
+import { SpacesService } from '~modules/spaces/services/spaces.service';
+
+@Injectable()
+export class SitemapService {
+	constructor(
+		protected dataService: DataService,
+		protected spacesService: SpacesService,
+		protected contentPagesService: ContentPagesService,
+		protected ieObjectsService: IeObjectsService,
+		protected assetsService: AssetsService
+	) {}
+
+	public async generateSitemap() {
+		const sitemapConfig = await this.getSitemapConfig();
+		const contentPagesPaths = await this.getContentPagesPaths();
+
+		const staticPages = ['/', '/bezoek', '/geheime-content-pagina']; // /geheime-content-pagina is for testing
+
+		const activeSpaces = await this.spacesService.findAll(
+			{ size: 1000, status: [VisitorSpaceStatus.Active] },
+			null
+		);
+
+		const publicObjects = await this.ieObjectsService.findObjectsForSitemap([
+			IeObjectLicense.PUBLIEK_METADATA_LTD,
+			IeObjectLicense.PUBLIEK_METADATA_ALL,
+		]);
+
+		// Create sitemap info for general pages
+		const generalPages: SitemapItemInfo[] = this.blacklistAndPrioritizePages(
+			[
+				...staticPages.map((path) => ({
+					loc: process.env.CLIENT_HOST + path,
+					changefreq: 'monthly',
+				})),
+				...contentPagesPaths.map((path) => ({
+					loc: process.env.CLIENT_HOST + path,
+					changefreq: 'monthly',
+				})),
+				// '/zoeken' is seperate because of the required order
+				{
+					loc: process.env.CLIENT_HOST + '/zoeken',
+					changefreq: 'monthly',
+				},
+				...activeSpaces.items.map((space) => ({
+					loc: process.env.CLIENT_HOST + '/zoeken/?aanbieders=' + space.maintainerId,
+					changefreq: 'weekly',
+					lastmod: moment(space.updatedAt).format('YYYY-MM-DD'),
+				})),
+			],
+			sitemapConfig
+		);
+
+		// Create sitemap info for item detail pages
+		const itemDetailPages: SitemapItemInfo[] = this.blacklistAndPrioritizePages(
+			[
+				...publicObjects.map((object) => ({
+					loc:
+						process.env.CLIENT_HOST +
+						'/zoeken/' +
+						object.maintainerSlug +
+						'/' +
+						object.schemaIdentifier,
+					changefreq: 'weekly',
+					lastmod: moment(object.updatedAt).format('YYYY-MM-DD'),
+				})),
+			],
+			sitemapConfig
+		);
+		const XML_OBJECTS_SIZE = 50000; // amount of objects in 1 xml file, should be 50000
```

Comment:
REQ: consts like these should be in a consts file so that they can be easily edited in one place.